src/qibolab/transpilers/routing.py

```diff
@@ -0,0 +1,339 @@
+
+        Returns ``True`` if the following conditions are satisfied:
+            - Circuit does not contain more than two-qubit gates.
+            - Circuit matches connectivity.
+            otherwise returns ``False``.
+        """
+        return respect_connectivity(connectivity=self.connectivity, circuit=circuit, verbose=self.verbose)
+
+    def __call__(self, circuit, initial_layout):
+        """Circuit connectivity matching.
+
+        Args:
+            circuit (:class:`qibo.models.Circuit`): circuit to be matched to hardware connectivity.
+            initial_layout (dict): initial qubit mapping.
+
+        Returns:
+            hardware_mapped_circuit (qibo.Circuit): circut mapped to hardware topology.
+            final_mapping (dict): final qubit mapping.
+        """
+        self._mapping = initial_layout
+        init_qubit_map = np.asarray(list(initial_layout.values()))
+        self.initial_checks(circuit.nqubits)
+        self._circuit_repr = create_circuit_repr(circuit)
+        self._graph = nx.relabel_nodes(self.connectivity, self._mapping)
+        self._qubit_map = np.sort(init_qubit_map)
+        self.first_transpiler_step(circuit)
+        while len(self._circuit_repr) != 0:
+            self.transpiler_step(circuit)
+        final_mapping = {
+            key: self._qubit_map[init_qubit_map[i]] for i, key in enumerate(list(self.connectivity.nodes()))
+        }
+        hardware_mapped_circuit = remap_circuit(self._transpiled_circuit, np.argsort(init_qubit_map))
+        return hardware_mapped_circuit, final_mapping
+
+    def transpiler_step(self, qibo_circuit):
+        """Transpilation step. Find new mapping, add swap gates and apply gates that can be run with this configuration.
+
+        Args:
+            qibo_circuit (:class:`qibo.models.Circuit`): circuit to be transpiled.
+        """
+        len_2q_circuit = len(self._circuit_repr)
+        path, meeting_point = self.relocate()
+        self.add_swaps(path, meeting_point)
+        self.update_qubit_map()
+        self.add_gates(qibo_circuit, len_2q_circuit - len(self._circuit_repr))
+
+    def first_transpiler_step(self, qibo_circuit):
+        """First transpilation step. Apply gates that can be run with the initial qubit mapping.
+
+        Args:
+            qibo_circuit (:class:`qibo.models.Circuit`): circuit to be transpiled.
+        """
+        self._circuit_position = 0
+        self.added_swaps = 0
+        len_2q_circuit = len(self._circuit_repr)
+        self._circuit_repr = self.reduce(self._graph)
+        self.add_gates(qibo_circuit, len_2q_circuit - len(self._circuit_repr))
+
+    @property
+    def sampling_split(self):
+        return self._sampling_split
+
+    @sampling_split.setter
+    def sampling_split(self, sampling_split):
+        """Set the sampling split.
+
+        Args:
+            sampling_split (float): define fraction of shortest path tested.
+        """
+
+        if sampling_split > 0.0 and 1.0 >= sampling_split:
```

Comment:
```suggestion
        if 1.0 >= sampling_split > 0.0:
```
I recently discovered that this is a possibility, should be easier to read and faster to write